scipy/spatial/transform/rotation.py

```diff
@@ -0,0 +1,862 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+import scipy.linalg
+import re
+import warnings
+
+
+AXIS_TO_IND = {'x': 0, 'y': 1, 'z': 2}
+
+
+def _elementary_basis_vector(axis):
+    b = np.zeros(3)
+    b[AXIS_TO_IND[axis]] = 1
+    return b
+
+
+def _compute_euler_from_dcm(dcm, seq, extrinsic=False):
+    # The algorithm assumes intrinsic frame transformations. For representation
+    # the paper uses transformation matrices, which are transpose of the
+    # direction cosine matrices used by our Rotation class.
+    # Adapt the algorithm for our case by
+    # 1. Instead of transposing our representation, use the transpose of the
+    #    O matrix as defined in the paper, and be careful to swap indices
+    # 2. Reversing both axis sequence and angles for extrinsic rotations
+
+    if extrinsic:
+        seq = seq[::-1]
+
+    if dcm.ndim == 2:
+        dcm = dcm[None, :, :]
+    num_rotations = dcm.shape[0]
+
+    # Step 0
+    # Algorithm assumes axes as column vectors, here we use 1D vectors
+    n1 = _elementary_basis_vector(seq[0])
+    n2 = _elementary_basis_vector(seq[1])
+    n3 = _elementary_basis_vector(seq[2])
+
+    # Step 2
+    sl = np.dot(np.cross(n1, n2), n3)
+    cl = np.dot(n1, n3)
+
+    # angle offset is lambda from the paper referenced in [2] from docstring of
+    # `as_euler` function
+    offset = np.arctan2(sl, cl)
+    c = np.vstack((n2, np.cross(n1, n2), n1))
+
+    # Step 3
+    rot = np.array([
+        [1, 0, 0],
+        [0, cl, sl],
+        [0, -sl, cl],
+    ])
+    res = np.einsum('...ij,...jk->...ik', c, dcm)
+    dcm_transformed = np.einsum('...ij,...jk->...ik', res, c.T.dot(rot))
+
+    # Step 4
+    angles = np.empty((num_rotations, 3))
+    angles[:, 1] = np.arccos(dcm_transformed[:, 2, 2])
```

Comment:
This place causes nans, sure about 99%. You need to ensure that `dcm_transformed[:, 2, 2]` doesn't exceed 1 by absolute value. Or you could assign these angle to 0 or pi depending on the sign. Probably the first approach is cleaner.